internal/httpx/httpx.go

```diff
@@ -142,14 +142,43 @@ func (c *apiClient) newRequestWithJSONBody(
 	return request, nil
 }
 
+// joinURLPath joins the BaseURL and the URLPath, allows httpx to access the URL
+func (c *apiClient) joinURLPath(URLPath string) string {
+
+	if c.BaseURL != "" && URLPath != "" {
+		if c.BaseURL[len(c.BaseURL)-1] != '/' && URLPath[0] != '/' {
+			c.BaseURL += "/"
+			return c.BaseURL + URLPath
+		}
+
+		if c.BaseURL[len(c.BaseURL)-1] == '/' && URLPath[0] == '/' {
+			return c.BaseURL + URLPath[1:]
+		}
+
+		if c.BaseURL[len(c.BaseURL)-1] == '/' && URLPath[0] != '/' {
+			return c.BaseURL + URLPath
+		}
+
+		if c.BaseURL[len(c.BaseURL)-1] != '/' && URLPath[0] == '/' {
+			return c.BaseURL + URLPath
+		}
+		return c.BaseURL + URLPath
+	}
+
+	if c.BaseURL != "" && URLPath == "" {
+		return c.BaseURL
+	}
+
+	return ""
+}
+
 // newRequest creates a new request.
 func (c *apiClient) newRequest(ctx context.Context, method, resourcePath string,
 	query url.Values, body io.Reader) (*http.Request, error) {
-	URL, err := url.Parse(c.BaseURL)
+	URL, err := url.Parse(c.joinURLPath(resourcePath))
 	if err != nil {
 		return nil, err
 	}
-	URL.Path = resourcePath
```

Comment:
I realize this is a matter of taste, but I have this preference. I'd like `joinURLPath` to take two URL paths and join them. I see this as slightly better than the current implementation. Hence my suggestion:

1. keep the previous code for parsing the URL, i.e., `URL, err := url.Parse(c.BaseURL)`
2. modify line 152 to be like

```Go
URL.Path = c.joinURLPath(URL.Path, resourcePath)
```

3. then the implementation of `joinURLPath` could be like

```Go
if !strings.HasSuffix(origPath, "/") {
  origPath += "/"
}
if strings.HasPrefix(newPath, "/") {
  newPath = newPath[1:]
}
return origPath + newPath
```

4. luckily you've written comprehensive tests, so we should ~quickly see whether this proposed code is still working as intended or not.

5. you should also update the tests as mentioned below

6. please, update the documentation to ensure that we mention this behavior

Reply:
Thanks for the code review, I was searching for built-in functions, but every google search ended up old school way 🤦🏻‍♂️